### PR TITLE
Category select badges

### PR DESCRIPTION
**Pull request by derpdead**
# Description

- Add support for informing user about selected subcategories via badges
- Improve code readability

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Unit Tests
- [ ] e2e Test

# Checklist:

- [x] I have read the contribution requirements and fulfil them.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: modules/@ergonode/categories/src/components/Selects/CategorySelect.vue [hunk 1]
```diff
@@ -393,6 +338,13 @@ export default {
 
             return [];
         },
+        visibleCategoryItems() {
+            if (!this.isCategoriesExpanded) {
+                return this.categoryItems.slice(0, 7);
```
Comment (bleto) [inline modules/@ergonode/categories/src/components/Selects/CategorySelect.vue:343]: `7` should not be in the default variable?

File: modules/@ergonode/categories/src/components/Selects/CategorySelectSelectAllCategories.vue [hunk 2]
```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE for license details.
+ */
+<template>
+    <CheckBox
+        class="category-select-select-all-categories"
```
Comment (bleto) [inline modules/@ergonode/categories/src/components/Selects/CategorySelectSelectAllCategories.vue:7]: `select-select` this duplication is ok ?  it looks bad

File: modules/@ergonode/categories/src/components/Selects/CategorySelectSelectAllCategories.vue [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            if (value) {
+                this.$emit('input', [
+                    ...new Set([
+                        ...this.value,
+                        ...this.categories.map(category => category.id),
+                    ]),
+                ]);
+            } else {
+                this.$emit('input', this.value.filter(id => !this.categories.some(category => category.id === id)));
+            }
```
Comment (bleto) [inline modules/@ergonode/categories/src/components/Selects/CategorySelectSelectAllCategories.vue:67]: Maybe something like this? 
```
let inputValue = [
                    ...new Set([
                        ...this.value,
                        ...this.categories.map(category => category.id),
                    ]),
                ]
if (!value) {
  inputValue = this.value.filter(id => !this.categories.some(category => category.id === id)));
}
this.$emit('input', inputValue);
```

File: modules/@ergonode/ui/src/models/treeAccordion/index.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    const count = (tree) => {
+        let sum = 0;
+
+        (tree.children || []).forEach((node) => {
+            sum += count(node);
+        });
+
+        const selectedValue = value.find(({
+            id,
+        }) => id === tree.id);
```
Comment (bleto) [inline modules/@ergonode/ui/src/models/treeAccordion/index.js:26]: if the order does not matter, move the variable declarations to the top of the function.

File: modules/@ergonode/ui/src/models/treeAccordion/index.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    ...prev,
+    [curr.id]: true,
+}), {});
+
+export const getSelectedNodesCount = ({
+    value,
+    treeStructure,
+}) => {
+    const nodesCount = {};
+
```
Comment (bleto) [inline modules/@ergonode/ui/src/models/treeAccordion/index.js:16]: no need to break the lines

File: modules/@ergonode/ui/src/models/treeAccordion/index.js (hunk 4 above)
Comment (derpdead) [reply modules/@ergonode/ui/src/models/treeAccordion/index.js:26]: it's reccurency, it's matter

File: modules/@ergonode/categories/src/components/Selects/CategorySelect.vue (hunk 1 above)
Comment (derpdead) [reply modules/@ergonode/categories/src/components/Selects/CategorySelect.vue:343]: It's already `visibleCategoryItems` => 7 is pointing how many of them are sliced.

File: modules/@ergonode/categories/src/components/Selects/CategorySelectSelectAllCategories.vue (hunk 3 above)
Comment (derpdead) [reply modules/@ergonode/categories/src/components/Selects/CategorySelectSelectAllCategories.vue:67]: It's kinda the same logic

File: modules/@ergonode/ui/src/models/treeAccordion/index.js (hunk 5 above)
Comment (derpdead) [reply modules/@ergonode/ui/src/models/treeAccordion/index.js:16]: I like to have function declaration  with one break

**Review by bleto — APPROVED** [review]

**Pull request merged**